openedx/features/course_experience/utils.py

```diff
@@ -35,29 +37,65 @@ def populate_children(block, all_blocks):
 
     def set_last_accessed_default(block):
         """
-        Set default of False for last_accessed on all blocks.
+        Set default of False for resume_block on all blocks.
         """
-        block['last_accessed'] = False
+        block['resume_block'] = False
+        block['complete'] = False
         for child in block.get('children', []):
             set_last_accessed_default(child)
 
+    def mark_blocks_completed(block):
+        """
+        Walk course tree, marking block completion.
+        Mark 'most recent completed block as 'resume_block'
+
+        """
+        def recurse_mark_complete(completion_query, last_complete, block):
+            locatable_block_string = BlockUsageLocator.from_string(block['id'])
+            if BlockUsageLocator.from_string(block['id']) in completion_query.keys():
+                block['complete'] = True
+                if locatable_block_string == last_complete.keys()[0]:
+                    block['resume_block'] = True
+
+            if block.get('children'):
+                for child in block['children']:
+                    block['children'][block['children'].index(child)] = recurse_mark_complete(
+                        completion_query,
+                        last_complete,
+                        block=child
+                    )
+                    if block['children'][block['children'].index(child)]['resume_block'] is True:
+                        block['resume_block'] = True
+
+            return block
+        last_completed_child_position = completion_service_instance.get_latest_block_completed()
```

Comment:
#NIT Should have a new line before this line to separate the function above from this statement

Reply:
👍